xlators/cluster/afr/src/afr-self-heal-common.c

```diff
@@ -161,7 +161,8 @@ afr_gfid_sbrain_source_from_src_brick(xlator_t *this, struct afr_reply *replies,
 
     priv = this->private;
     for (i = 0; i < priv->child_count; i++) {
-        if (!replies[i].valid || replies[i].op_ret == -1)
+        if (!replies[i].valid || replies[i].op_ret == -1 ||
+            afr_is_any_outcast_set(replies[i].xdata))
```

Comment:
All these changes in self-heal logic can also happen from clients (if client side heal is enabled). So if old clients are also accessing the volume, the problem can still occur no?